src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -270,6 +270,14 @@ public virtual void Contextualize(ViewContext viewContext)
         /// <inheritdoc />
         public IHtmlContent AntiForgeryToken()
         {
+            // If we're inside a BeginForm/BeginRouteForm, the antiforgery token might have already been
+            // created and appended to the 'end form' content.
+            if (ViewContext.FormContext.HasAntiforgeryToken)
+            {
+                return HtmlString.Empty;
+            }
```

Comment:
Slightly odd to give precedence to the often-implicit token from `BeginForm()` / `BeginRouteForm()`. Any known cases where user code either depends on the hidden field's location in the `<form>` or makes temporary changes to `ViewContext` when calling `AntiForgeryToken()`?